props/saber_fett263_buttons.h

```diff
@@ -293,248 +659,3857 @@ class SaberFett263Buttons : public virtual PropBase {
+            char igt[32];
+            style_parser.GetArgument(current_preset_.GetStyle(blade_num_), 19, igt);
+            ignite_time_ = strtol(igt, NULL, 0);
+            current_preset_.SetStyle(blade_num_,style_parser.SetArgument(current_preset_.GetStyle(blade_num_), 19, "1"));
+            arg_revert_ = strtol(argspace, NULL, 0);
+            break;
+          case 21:
+            PlayMenuSound("mselect.wav");
+            menu_type_ = MENU_CLASH_LOCATION;
+            ShowColorStyle::SetColor(GetColorArg(blade_num_, 4));
+            show_clash_location_.Start(blade_num_);
+            break;
+          case 22:
+            PlayMenuSound("mselect.wav");
+            menu_type_ = MENU_DRAG_SIZE;
+            ShowColorStyle::SetColor(GetColorArg(blade_num_, 6));
+            show_drag_size_.Start(blade_num_);
+            break;
+          case 23:
+            PlayMenuSound("mselect.wav");
+            menu_type_ = MENU_EMITTER_SIZE;
+            ShowColorStyle::SetColor(GetColorArg(blade_num_, 1));
+            show_emitter_size_.Start(blade_num_);
+            break;
+          case 24:
+            PlayMenuSound("moption.wav");
+            if (SFX_preon) {
+              menu_type_ = MENU_PREON_OPTION;
+              arg_revert_ = strtol (argspace, NULL, 0);
+            }
+            break;
+          case 25:
+            if (!SFX_preon) {
+              PlayMenuSound("moption.wav");
+            } else {
+              PlayMenuSound("mselect.wav");
+              menu_type_ = MENU_PREON_SIZE;
+              arg_revert_ = strtol (argspace, NULL, 0);
+            }
+            break;
+          default:
+            break;
+        }
+        break;
+      case MENU_STYLE_OPTION:
+      case MENU_IGNITION_OPTION:
+      case MENU_IGNITION_TIME:
+      case MENU_PREON_OPTION:
+      case MENU_PREON_SIZE:
+         menu_type_ = MENU_STYLE_SETTING_SUB;
+         current_preset_.Save();
+         MenuSave();
+         break;
+      case MENU_RETRACTION_OPTION:
+      case MENU_RETRACTION_TIME:
+         menu_type_ = MENU_STYLE_SETTING_SUB;
+         char style_arg[10];
+         itoa(ignite_time_, style_arg, 10);
+         current_preset_.SetStyle(blade_num_,style_parser.SetArgument(current_preset_.GetStyle(blade_num_), 19, style_arg));
+         current_preset_.Save();
+         MenuSave();
+         break;
+      case MENU_CLASH_LOCATION:
+      case MENU_DRAG_SIZE:
+      case MENU_EMITTER_SIZE:
+        SaveStyleSetting();
+        break;
+      case MENU_DELETE:
+        if (choice_) {
+          current_preset_.SaveAt(-1);
+          choice_ = false;
+          menu_type_ = MENU_TOP;
+          MenuSelect();
+          break;
+        }
+        PlayMenuSound("maffirm.wav");
+        break;
+      case MENU_EXIT:
+        PlayMenuSound("mselect.wav");
+        MenuExit();
+        break;
+  #endif
+      default:
+        break;  
+      }
+  }
+
+// Edit Mode Dial Right      
+  void MenuDialRight() {
```

Comment:
As far as I can tell, MenuDialLeft and MenuDialRight are very very similar.
In a lot of cases, the only differences are:

right:

 track_num_ += 1;
 if (track_num_ > num_tracks_ - 1) track_num_ = 0;

left:

track_num_ -= 1;
if (track_num_ < 0) track_num_ = num_tracks_ -1;

I think we should combine them into a single function, like:

void MenuDial(int direction); 

direction will be -1 for right and 1 for left.

Then the code above becomes:
track_num_ += direction;
if (track_num_ >= num_tracks_) track_num_ = 0;
if (track_num_ < 0) track_num_ = num_tracks_ -1;

Reply:
consolidated to MenuDial()